scipy/special/boost_special_functions.h

```diff
@@ -760,46 +760,93 @@ ncf_pdf_double(double x, double v1, double v2, double l)
 
 template<typename Real>
 Real
-ncf_cdf_wrap(const Real x, const Real v1, const Real v2, const Real l)
+ncf_cdf_wrap(const Real v1, const Real v2, const Real l, const Real x)
 {
-    if (std::isfinite(x)) {
-        return boost::math::cdf(
-            boost::math::non_central_f_distribution<Real, StatsPolicy>(v1, v2, l), x);
+    if (std::isnan(x) || std::isnan(v1) || std::isnan(v2) || std::isnan(l)) {
+	return NAN;
     }
-    // -inf => 0, inf => 1
-    return 1.0 - std::signbit(x);
+    if ((v1 <= 0) || (v2 <= 0) || (l < 0) || (x < 0)) {
+	sf_error("ncfdtr", SF_ERROR_DOMAIN, NULL);
+	return NAN;
+    }
+    if (std::isinf(x)) {
+	// -inf => 0, inf => 1
+	return 1.0 - std::signbit(x);
```

Comment:
We've already returned if `x < 0`, so all that's left is ` x == inf`, right?

Reply:
Yes. Good catch.  A holdover from this once having the natural cdf behavior because it was copied from the scalar kernel for `_ncf_cdf`.